examples/generative/corrdiff/generate.py

```diff
@@ -173,13 +174,24 @@ def main(cfg: DictConfig) -> None:
     # move to device
     device = dist.device
     net_res = net_res.eval().to(device).to(memory_format=torch.channels_last)
-    net_reg = net_reg.eval().to(device).to(memory_format=torch.channels_last) if net_reg else None
+    net_reg = (
+        net_reg.eval().to(device).to(memory_format=torch.channels_last)
+        if net_reg
+        else None
+    )
 
     # change precision if needed
     if force_fp16:
         net_reg.use_fp16 = True
         net_res.use_fp16 = True
 
+    # Reset since we are using a different mode.
+    torch._dynamo.reset()
+    compile_mode = "reduce-overhead"
+    # Only compile residual network
+    # Overhead of compiling regression network outweights any benefits
+    net_res = torch.compile(net_res, mode=compile_mode)
+
```

Comment:
Can you specify what version of pytorch or container you tested this on? I have was having issues with 2.0 vs. 2.1.

Reply:
This was with `torch.__version__ = '2.1.0a0+32f93b1'`

Reply:
Does it make sense to make compile mode configurable?

Reply:
I agree because I think the necessary `torch.compile` support is not available in `2.0`

From the release notes:
`torch.compile can now compile NumPy operations via translating them into PyTorch-equivalent operations.`

Reply:
Yes, makes sense. I'll add that configuration option.